packages/flutter/lib/src/material/pickers/date_picker_dialog.dart

```diff
@@ -383,11 +399,15 @@ class _DatePickerDialogState extends State<_DatePickerDialog> {
       layoutBehavior: ButtonBarLayoutBehavior.constrained,
       children: <Widget>[
         FlatButton(
-          child: Text(widget.cancelText ?? localizations.cancelButtonLabel),
+          child: Text(widget.cancelText ?? localizations.cancelButtonLabel,
+            key: _cancelButtonKey
```

Comment:
Formatting seems a bit off here, perhaps it makes more sense to add the key to the `FlatButton` instead?

Same with the OK button

Reply:
This had to do with the key'd widget needing to be inside the InkWell that handled the activation. I have decided that this is not a great way to handle this and removed it from this PR. I have a better way to more generically handle dismissing modal routes with a 'escape' that I will talk with Greg about and put together another PR.